instrumentation/jaxrs/jaxrs-common/testing/src/main/java/io/opentelemetry/instrumentation/jaxrs/AbstractJaxRsHttpServerTest.java

```diff
@@ -0,0 +1,573 @@
+    testKind.assertBody(response.contentUtf8());
+
+    /*
+    def spanCount = 2
+    def hasSendError = asyncCancelHasSendError() && action == "cancel"
+    if (hasSendError) {
+      spanCount++
+    }
+    assertTraces(1) {
+      trace(0, spanCount) {
+        asyncServerSpan(it, 0, url, statusCode)
+        handlerSpan(it, 1, span(0), "asyncOp", isCancelled, isError, errorMessage)
+        if (hasSendError) {
+          sendErrorSpan(it, 2, span(1))
+        }
+      }
+    }
+     */
+
+    testing()
+        .waitAndAssertTraces(
+            trace -> {
+              List<Consumer<SpanDataAssert>> assertions =
+                  new ArrayList<>(
+                      Arrays.asList(
+                          span ->
+                              assertServerSpan(
+                                  span,
+                                  "GET",
+                                  new ServerEndpoint(
+                                      "async",
+                                      "async?action=" + testKind.action,
+                                      testKind.statusCode,
+                                      null,
+                                      false),
+                                  testKind.statusCode),
+                          span -> {
+                            span.hasName("JaxRsTestResource.asyncOp")
+                                .hasKind(SpanKind.INTERNAL)
+                                .hasParent(trace.getSpan(0))
+                                .hasAttributesSatisfyingExactly(
+                                    satisfies(
+                                        CodeIncubatingAttributes.CODE_NAMESPACE,
+                                        name -> name.endsWith("JaxRsTestResource")),
+                                    equalTo(CodeIncubatingAttributes.CODE_FUNCTION, "asyncOp"),
+                                    equalTo(
+                                        AttributeKey.booleanKey("jaxrs.canceled"),
+                                        testKind == AsyncResponseTestKind.CANCELED ? true : null));
+                            if (testKind == AsyncResponseTestKind.FAILING) {
+                              span.hasStatus(StatusData.error())
+                                  .hasException(new IllegalStateException("failure"));
+                            }
+                          }));
+              if (asyncCancelHasSendError() && testKind == AsyncResponseTestKind.CANCELED) {
+                assertions.add(
+                    span ->
+                        span.satisfies(
+                                spanData -> assertThat(spanData.getName()).endsWith("sendError"))
+                            .hasKind(SpanKind.INTERNAL)
+                            .hasParent(trace.getSpan(1)));
+              }
+              trace.hasSpansSatisfyingExactly(assertions);
+            });
+  }
+
+  /*
+
+  @Unroll
+  def "should handle #desc AsyncResponse"() {
```

Comment:
there's several groovy snippets commented out that that were probably forgotten to be removed

Reply:
thanks